### ISSUE TITLE
Add support for common parameters defined on an OAS path item

### ISSUE THREAD

**Issue by austinvalle** [hashicorp/terraform-plugin-codegen-openapi#115]
### Use Cases or Problem Statement

Currently, `tfplugingen-openapi` only detects parameters that are defined in the [Operation](https://spec.openapis.org/oas/latest.html#operation-object) object (`GET` method) of an OpenAPI spec. [Common parameters](https://swagger.io/docs/specification/describing-parameters/#common-for-path) for **all** operations can be defined in the parent [Path Item](https://spec.openapis.org/oas/latest.html#path-item-object) object.

```yaml
paths:
  /user/{id}:
    parameters:
      - in: path
        name: id
        schema:
          type: integer
        required: true
        description: The user ID
    get:
      summary: Gets a user by ID
      ...
    patch:
      summary: Updates an existing user with the specified ID
      ...
    delete:
      summary: Deletes the user with the specified ID
      ...
```

### Proposal

These common parameters defined in path item, should be combined with the existing operation object parameters:

```yaml
paths:
  /users/{id}:
    parameters:
      - in: path
        name: id
        schema:
          type: integer
        required: true
        description: The user ID.
    # GET/users/{id}?metadata=true
    get:
      summary: Gets a user by ID
      # Note we only define the query parameter, because the {id} is defined at the path level.
      parameters:
        - in: query
          name: metadata
          schema:
            type: boolean
          required: false
          description: If true, the endpoint returns only the user metadata.
      responses:
        '200':
          description: OK
```
### Note
- We should also support overriding any common parameters with operation level parameters, per the [OpenAPI specification](https://spec.openapis.org/oas/latest.html#path-item-object):

> _A list of parameters that are applicable for all the operations described under this path. These parameters can be overridden at the operation level, but cannot be removed there. The list MUST NOT include duplicated parameters. A unique parameter is defined by a combination of a name and location._

- We also should determine how to handle a query parameter name that has a conflict with a path parameter name. This would be valid in OpenAPI, but would create an invalid Terraform schema with duplicate attributes.

### Additional Information

- Original proposal and work in #114 by @randomswdev

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

**Comment by austinvalle** [comment]
These changes have been merged to `main` and will be released with `v0.3.0` sometime later this week. As always, if you need the changes ASAP you can `go install` from the latest commit in `main`

```bash
go install github.com/hashicorp/terraform-plugin-codegen-openapi/cmd/tfplugingen-openapi@dfe71df55db5285bcb904e2d6262ff62fd2be3df
```

**Comment by austinvalle** [comment]
[`v0.3.0`](https://github.com/hashicorp/terraform-plugin-codegen-openapi/releases/tag/v0.3.0) released 🥳
